lib/graphql/batch/executor.rb

```diff
@@ -35,5 +36,21 @@ def wait_all
     def clear
       loaders.clear
     end
+
+    def defer
```

Comment:
Note that if someone wanted to override this method and actually defer the callback, then the callback would get called outside of the parent `with_loading` block.

I'll add some comments to the code to clarify the purpose of `loading` outside this PR, and mention why `defer` is overridden here.